nilearn/signal.py

```diff
@@ -641,3 +562,136 @@ def clean(signals, sessions=None, detrend=True, standardize='zscore',
                                detrend=False)
 
     return signals
+
+
+def _filter_signal(signals, confounds, filter, low_pass, high_pass, t_r):
+    '''Filter signal based on provided strategy.'''
+    if filter == 'butterworth':
+        signals = butterworth(signals, sampling_rate=1. / t_r,
+                              low_pass=low_pass, high_pass=high_pass)
+        if confounds is not None:
+            # Apply low- and high-pass filters to keep filters orthogonal
+            # (according to Lindquist et al. (2018))
+            confounds = butterworth(confounds, sampling_rate=1. / t_r,
+                                    low_pass=low_pass, high_pass=high_pass)
+    elif filter == "cosine":
+        from .glm.first_level.design_matrix import _cosine_drift
+        frame_times = np.arange(signals.shape[0]) * t_r
+        cosine_drift = _cosine_drift(high_pass, frame_times)
+        if confounds is None:
+            confounds = cosine_drift.copy()
+        else:
+            confounds = np.hstack((confounds, cosine_drift))
+    return signals, confounds
+
+
+
+def _process_runs(signals, runs, detrend, standardize, confounds, low_pass, high_pass, t_r):
+    """Process each run independently."""
+    if len(runs) != len(signals):
+        raise ValueError(('The length of the session vector (%i) '
+                            'does not match the length of the signals (%i)')
+                            % (len(runs), len(signals)))
+    for s in np.unique(runs):
```

Comment:
```suggestion
    for run in np.unique(runs):
```(nitpick, but this is more readable)